var/spack/repos/builtin/packages/specfem3d-globe/package.py

```diff
@@ -22,6 +23,9 @@ class Specfem3dGlobe(AutotoolsPackage, CudaPackage):
             description="Build with OpenMP code generator")
     variant('double-precision', default=False,
             description="Treat REAL as double precision")
+    variant('par_file', values=str, default='original',
+            description="Specify the 'Par_file': parameter file "
+                        "you want to use with the full path")
```

Comment:
@alalazo 
`Par_file` is parameter configuration file involved in the `specfem3d-globe` package. 
This file has setting items such as `SIMULATION_TYPE`, and user is free to change its settings.
`specfem3d-globe` will generate an optimized binary based on the settings in this file.

If we want to use different settings, we need to replace the `Par_file` and rebuild.
So it is effective to have a separate hash to keep the binary of each setting.

Reply:
@alalazo Is the above explanation sufficient? If not, please contact me.

Reply:
Thanks for the explanation.  Having a file like that seems to defeat Spack hashing mechanism:
1. Spack will encode the absolute path of the file in the hash
2. The file can be modified and the executable rebuilt. The hash will be the same, the binaries will not.
3. Any relocation would lead to either an inconsistent state (variant referring to a path that may not be on the machine we deploy) or to a change in hash

Can you link some docs to this file? If there are not a lot of parameters we may generate the file during the configuration and just expose some handles.

Reply:
@alalazo 
Sorry for the late response... This is the Par_file that specfem3d_globe has.
https://github.com/geodynamics/specfem3d_globe/blob/master/DATA/Par_file
This file has many parameters, so we specified the file in a variant.